### PR TITLE
change the flow for students to access the quiz by code

### PR DESCRIPTION
**Pull request by naderakhgari**
Thank you for contributing! To help the review process, please provide the following:

### Proposal
change the flow for students to access the quiz by code

### Related
Provide links to any associated pull requests or issues. You can use `#123` to link to a PR or issue in this repository, or `user/repo#123` to link to other repositories.

### Checklist

- [ ] I have read the [code of conduct] and [contributing guide]
- [ ] I have made this pull request to the `master` branch
- [ ] I have run all of the automated validation using `npm run ship`
- [ ] I have added myself to the `"contributors"` list in the `package.json` (or do not want to)

[code of conduct]: https://github.com/textbook/starter-kit/blob/master/.github/CODE_OF_CONDUCT.md
[contributing guide]: https://github.com/textbook/starter-kit/blob/master/.github/CONTRIBUTING.md

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -34,29 +37,41 @@ const App = () => {
...
+            New question
 					</Link>
 				</nav>
 				<div className="body">
 					<Switch>
 						<Route exact path="/Mentors">
-							<Mentors questions={questions} quizes={quizzes} />
+							<Mentors
+								questions={questions}
+								quizes={quizzes}
```
Comment (pendletons) [inline client/src/App.js:57]:
```suggestion
								quizzes={quizzes}
```

File: client/src/Components/Mentors.js [hunk 2]
```diff
@@ -125,20 +129,38 @@ export default function Mentors(props) {
 	if (filteredQuestionsByTag) {
 		return (
 			<div className="row">
-				<div className='filterButtons col-6'>
-
-					<button onClick={ autofillQuizz }>autofill quiz</button>
+				<div className="filterButtons col-6">
+					<RunQuiz
+						quizzes={props.quizes}
```
Comment (pendletons) [inline client/src/Components/Mentors.js:134]:
```suggestion
						quizzes={props.quizzes}
```

File: client/src/Components/Mentors.js [hunk 3]
```diff
@@ -125,20 +129,38 @@ export default function Mentors(props) {
...
+						setRoute={props.setRoute}
+						setCode={props.setCode}
+						code={props.code}
+						quizes={props.quizzes}
```
Comment (pendletons) [inline client/src/Components/Mentors.js:138]:
```suggestion
						quizzes={props.quizzes}
```

File: client/src/Components/RunQuiz.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+		props.setQuizId(event.target.value);
+		props.setCode(
+			Math.random()
+				.toString(36)
+				.replace(/[^a-z0-9]+/g, "")
+				.substr(0, 3)
+		);
+	};
+
+	console.log(props.quizzes);
```
Comment (pendletons) [inline client/src/Components/RunQuiz.js:15]: Probably don't want to keep this in ;-)

**Review by pendletons — APPROVED** [review]
Looks good, just a few tidying things.

File: client/src/Components/RunQuiz.js (hunk 4 above)
Comment (naderakhgari) [reply client/src/Components/RunQuiz.js:15]: yes
thanks for your reviews

**Review by sabia-m — APPROVED** [review]

**Pull request merged**